app/org/maproulette/framework/mixins/SearchParametersMixin.scala

```diff
@@ -0,0 +1,495 @@
+                | jsonb_array_elements(geojson->'features') features
+                | WHERE parent_id IN (${l.mkString(",")})
+                | AND (${tps.toSQL}))""".stripMargin)
+            CustomParameter(query.toString())
+          case _ =>
+            params.taskParams.taskProperties match {
+              case Some(tp) =>
+                val searchType = params.taskParams.taskPropertySearchType.getOrElse("equals")
+
+                val query = new StringBuilder(s"""tasks.id IN (
+                    | SELECT id FROM tasks,
+                    | jsonb_array_elements(geojson->'features') features
+                    | WHERE parent_id IN (${l.mkString(",")})
+                    | AND (true""".stripMargin)
+                for ((k, v) <- tp) {
+                  searchType match {
+                    case SearchParameters.TASK_PROP_SEARCH_TYPE_EQUALS =>
+                      query ++= s" AND features->'properties'->>'${k}' = '${v}' "
+                    case SearchParameters.TASK_PROP_SEARCH_TYPE_NOT_EQUAL =>
+                      query ++= s" AND features->'properties'->>'${k}' != '${v}' "
+                    case SearchParameters.TASK_PROP_SEARCH_TYPE_CONTAINS =>
+                      query ++= s" AND features->'properties'->>'${k}' LIKE '%${v}%' "
+                    case SearchParameters.TASK_PROP_SEARCH_TYPE_EXISTS =>
+                      query ++= s" AND features->'properties'->>'${k}' IS NOT NULL "
+                    case SearchParameters.TASK_PROP_SEARCH_TYPE_MISSING =>
+                      query ++= s" AND features->'properties'->>'${k}' IS NULL "
+                    case _ => // should not happen
+                  }
+                }
+                query ++= "))"
+                CustomParameter(query.toString())
+              case _ => CustomParameter("")
+            }
+        }
+      case None => CustomParameter("")
+    }
+  }
+
+  /**
+    * Filters on tasks task_review.review_requested_by
+    * @param params with inverting on 'o'
+    */
+  def filterOwner(params: SearchParameters): Parameter[String] = {
+    params.owner match {
+      case Some(o) if o.nonEmpty =>
+        val invert = if (params.invertFields.getOrElse(List()).contains("o")) "NOT" else ""
+        CustomParameter(s"""(tasks.id ${invert} IN
+            | (SELECT task_id
+            | FROM task_review tr
+            | INNER JOIN users u ON u.id = tr.review_requested_by
+            | WHERE tr.task_id = tasks.id AND LOWER(u.name) LIKE LOWER('%${o}%') ))""".stripMargin)
+      case _ => CustomParameter("")
+    }
+  }
+
+  /**
+    * Filters on tasks task_review.reviewed_by
+    * @param params with inverting on 'r'
+    */
+  def filterReviewer(params: SearchParameters): Parameter[String] = {
+    params.reviewer match {
+      case Some(r) if r.nonEmpty =>
+        val invert = if (params.invertFields.getOrElse(List()).contains("r")) "NOT" else ""
+        CustomParameter(s"""(tasks.id ${invert} IN
```

Comment:
This can be done a little differently to keep in style with our little DSL, kind of like this:
```scala
SubQueryFilter("tasks.id", Query.simple(
    List(
         BaseParameter("tr.task_id", "tasks.id", useValueDirectly = true), 
         BaseParameter("u.name", s"'%${r}%'", Operator.ILIKE)
    ), 
    "SELECT task_id FROM task_review tr INNER JOIN users u ON u.id = tr.reviewed_by"
)
```

I didn't check the syntax and whether that is exactly correct, but that is the general idea. One thing to watch out for is the table names. There might be an implicit table name being passed down and then appended to all the column names. One thing you might want to update as well is in the Parameter class there is a function called "getColumnKey" which basically sets the column name, it might be useful to check if there is a period in the string and if there is, then don't prepend the provided tableKey.

Reply:
Updated all three methods that do that subquery.